bee-node/build.rs

```diff
@@ -1,15 +1,58 @@
-// Copyright 2020-2021 IOTA Stiftung
+// Copyright 2020-2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use std::process::Command;
+use std::{fmt, process::Command};
 
+#[allow(dead_code)]
 #[derive(Debug)]
 enum BuildError {
-    GitCommit,
+    DashboardDecode,
+    DashboardDownload,
+    DashboardInvalidArchive,
+    DashboardNoSuitableRelease,
+    DashboardRequest(Option<u16>, String),
     GitBranch,
+    GitCommit,
+}
+
+impl fmt::Display for BuildError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match &self {
+            Self::DashboardDecode => write!(f, "failed to decode response from JSON"),
+            Self::DashboardDownload => write!(f, "failed to download latest release archive"),
+            Self::DashboardInvalidArchive => write!(f, "failed to open or extract release archive"),
+            Self::DashboardNoSuitableRelease => write!(f, "no release assets found for latest version"),
+            Self::DashboardRequest(Some(code), url) => write!(f, "failed request to `{}`: status code {}", url, code),
+            Self::DashboardRequest(_, url) => write!(f, "failed request to `{}`", url),
+            Self::GitBranch => write!(f, "failed to retrieve git branch name"),
+            Self::GitCommit => write!(f, "failed to retrieve git commit"),
+        }
+    }
 }
 
+#[cfg(not(feature = "dashboard"))]
 fn main() -> Result<(), BuildError> {
+    parse_git_information()
+}
+
+#[cfg(feature = "dashboard")]
+#[tokio::main]
+async fn main() -> Result<(), BuildError> {
+    parse_git_information()?;
+
+    let should_fetch = std::env::var("FETCH_DASHBOARD").map(|val| val == "1").unwrap_or(false);
```

Comment:
What is the motivation behind the `env` variable? Shouldn't `#[cfg(feature = "dashboard")]` be enough?

Reply:
Just pushed a couple of changes -- `FETCH_DASHBOARD` is there to force a rebuild/redownload of the release, if you wanted to update or redownload it without changing the code in the crate. The new change adds the `cargo:rerun-if-env-changed` to do this properly.